app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationsRepository.kt

```diff
@@ -0,0 +1,100 @@
+package com.keylesspalace.tusky.components.conversation
+
+import androidx.annotation.MainThread
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.paging.Config
+import androidx.paging.toLiveData
+import com.google.gson.Gson
+import com.keylesspalace.tusky.db.AppDatabase
+import com.keylesspalace.tusky.entity.Conversation
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.util.Listing
+import com.keylesspalace.tusky.util.NetworkState
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.util.concurrent.Executors
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ConversationsRepository @Inject constructor(val mastodonApi: MastodonApi, val db: AppDatabase, val gson: Gson) {
+
+    private val ioExecutor = Executors.newSingleThreadExecutor()
+
+    companion object {
+        private const val DEFAULT_PAGE_SIZE = 20
+    }
+
+    @MainThread
+    private fun refresh(accountId: Long): LiveData<NetworkState> {
+        val networkState = MutableLiveData<NetworkState>()
+        networkState.value = NetworkState.LOADING
+
+        mastodonApi.getConversations(null, DEFAULT_PAGE_SIZE).enqueue(
```

Comment:
I find vanilla requests super weird and I would rather replace this whooole construct with:
1. API returning Single
2. `networkState` as `PublishSubject`
3. `mastodonApi.getConversations(...).map {...}.onErrorReturnItem(...).subscribe(subject::onNext)`

There are automatic converters from Rx to LiveData but I cannot really fix everything in my head now and how much LiveData we really need

Reply:
I tried to, its not as easy as it seems (at least to me). Feel free to improve it later.